collector/generatorreceiver/internal/generator/trace_generator.go

```diff
@@ -136,26 +136,20 @@ func (g *TraceGenerator) createSpanForServiceRouteCall(traces *pdata.Traces, ser
 		}
 	}
 
-	maxEndTime := startTimeMicros
+	// TODO: this is still a bit weird - we're calling each downstream route
+	// after a sample of the current route's latency, which doesn't really
+	// make sense - but maybe it's realistic enough?
```

Comment:
Yeah, still weird but IMO the spans look pretty realistic. Maybe later on we could do some fancy math to spread the current route's latency between/after its downstream calls instead of sampling only beforehand, but I'm not sure how that would look exactly.

Reply:
Ooh, I like that